Bio/AlignIO/MafIO.py

```diff
@@ -0,0 +1,774 @@
+                    "Provided database indexed for %s, expected %s" % (
+                        db_target, self._target_seqname))
+
+            record_count = int(self._db_con.execute(
+                "SELECT value FROM meta_data WHERE key = 'record_count'").fetchone()[0])
+            if record_count == -1:
+                raise ValueError("Unfinished/partial database provided")
+
+            records_found = int(self._db_con.execute(
+                "SELECT COUNT(*) FROM offset_data").fetchone()[0])
+            if records_found != record_count:
+                raise ValueError(
+                    "Expected %s records, found %s.  Corrupt index?" % (
+                        record_count, records_found))
+
+            return records_found
+        except (_OperationalError, _DatabaseError) as err:
+            raise ValueError("Problem with SQLite database: %s" % err)
+
+    def __make_new_index(self):
+        """Read MAF file and generate SQLite index"""
+
+        from itertools import islice
+
+        # make the tables
+        self._db_con.execute("CREATE TABLE meta_data (key TEXT, value TEXT);")
+        self._db_con.execute("INSERT INTO meta_data (key, value) VALUES ('version', 1);")
+        self._db_con.execute("INSERT INTO meta_data (key, value) VALUES ('record_count', -1);")
+        self._db_con.execute(
+            "INSERT INTO meta_data (key, value) VALUES ('target_seqname', '%s');" % (
+                self._target_seqname,))
+        self._db_con.execute(
+            "INSERT INTO meta_data (key, value) VALUES ('filename', '%s');" % (self._maf_file,))
+        self._db_con.execute(
+            "CREATE TABLE offset_data (bin INTEGER, start INTEGER, end INTEGER, offset INTEGER);")
+
+        insert_count = 0
+
+        # iterate over the entire file and insert in batches
+        mafindex_func = self.__maf_indexer()
+
+        while True:
+            batch = list(islice(mafindex_func, 100))
+            if not batch:
+                break
+
+            # batch is made from self.__maf_indexer(),
+            # which yields zero-based "inclusive" start and end coordinates
+            self._db_con.executemany(
+                "INSERT INTO offset_data (bin, start, end, offset) VALUES (?,?,?,?);", batch)
+            self._db_con.commit()
+
+            insert_count += len(batch)
+
+        # then make indexes on the relevant fields
+        self._db_con.execute("CREATE INDEX IF NOT EXISTS bin_index ON offset_data(bin);")
+        self._db_con.execute("CREATE INDEX IF NOT EXISTS start_index ON offset_data(start);")
+        self._db_con.execute("CREATE INDEX IF NOT EXISTS end_index ON offset_data(end);")
+
+        self._db_con.execute(
+            "UPDATE meta_data SET value = '%s' WHERE key = 'record_count'" % (insert_count,))
+
+        self._db_con.commit()
+
+        return insert_count
+
+    def __maf_indexer(self):
+        """Generator function, yields index information for each bundle. in the
```

Comment:
Ideally we follow PEP257 for the doctoring format, i.e. online summary, blank line, details
pep257 d100